src/silx/gui/plot/ImageView.py

```diff
@@ -843,8 +857,8 @@ def setProfileWindowBehavior(self, behavior: str):
     @docstring(ImageView)
     def setImage(self, image, *args, **kwargs):
         if hasattr(image, 'dtype') and hasattr(image, 'shape'):
-            assert len(image.shape) == 2
-            height, width = image.shape
+            assert image.ndim in (2, 3)
```

Comment:
Maybe catch early RGB(A) without the right number of channels:
```suggestion
            assert image.ndim == 2 or (image.ndim == 3 and image.shape[2] in (3, 4))
```